src/ray/raylet/worker_pool.h

```diff
@@ -406,14 +457,11 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
     /// A map from the pids of starting worker processes
     /// to the number of their unregistered workers.
     std::unordered_map<Process, int> starting_worker_processes;
-    /// A map for looking up the task with dynamic options by the pid of the pending
+    /// A map for looking up the task with dynamic options by the pid of
     /// worker. Note that this is used for the dedicated worker processes.
-    std::unordered_map<Process, TaskID> pending_dedicated_workers_to_tasks;
+    std::unordered_map<Process, TaskID> dedicated_workers_to_tasks;
```

Comment:
Can you explain a little bit about why pending is removed now?

Reply:
pending_dedicated_workers wasn't present in the original code for dynamic_options; I recently added it as a workaround to prevent these dedicated workers from being returned to the general idle pool, because workers with environment variable overrides were (mistakenly) using the same code path as actors with dynamic_options.  Now that override_environment_variables is handled correctly in the WorkerCacheKey, the workaround is no longer needed.